applications/vila_live/vila_live.py

```diff
@@ -104,28 +159,47 @@ def __init__(self, data, source="v4l2", video_device="none"):
     def compose(self):
         pool = UnboundedAllocator(self, name="pool")
 
-        # V4L2 to capture usb camera input or replayer to replay video
-        if self.source == "v4l2":
-            v4l2_args = self.kwargs("v4l2_source")
+        # Sources initialization
+        sources = []
+        source_outputs = []
+
+        # Try sources with sequential numbering
+        for i in range(3):  # Start with trying v4l2_source_0 and v4l2_source_1
```

Comment:
Can we make the number of sources dependent on the config.yaml file as opposed to hardcoding to always use 3? This way using a single source is still an option.